v2/tools/generator/internal/functions/get_extended_resources_function.go

```diff
@@ -0,0 +1,110 @@
+	var packageRefs []astmodel.PackageReference
+	// Package reference for return type
+	packageRefs = append(packageRefs, astmodel.KubernetesResourceType.PackageReference)
+
+	for _, typeDef := range resources {
+		packageRefs = append(packageRefs, typeDef.PackageReference)
+	}
+	return packageRefs
+}
+
+//Sort resources according to the package name and resource names
+func sortResources(resources []astmodel.TypeName) []astmodel.TypeName {
+	sort.Slice(resources, func(i, j int) bool {
+		iVal := resources[i]
+		jVal := resources[j]
+
+		return iVal.PackageReference.PackageName() < jVal.PackageReference.PackageName() ||
+			iVal.PackageReference.PackageName() < jVal.PackageReference.PackageName() && iVal.Name() < jVal.Name()
+	})
+	return resources
+}
+
+// Name returns the name of this function, which is always GetExtendedResources()
+func (ext *GetExtendedResourcesFunction) Name() string {
+	return "GetExtendedResources"
+}
+
+// RequiredPackageReferences returns the set of packages required by GetExtendedResources()
+func (ext *GetExtendedResourcesFunction) RequiredPackageReferences() *astmodel.PackageReferenceSet {
+	return astmodel.NewPackageReferenceSet(getPackageRefs(ext.resources)...)
+}
+
+// References shows that GetExtendedResources() references nextother generated types
+func (ext *GetExtendedResourcesFunction) References() astmodel.TypeNameSet {
+	return astmodel.NewTypeNameSet()
+}
+
+// AsFunc returns the generated code for the GetExtendedResources() function
+func (ext *GetExtendedResourcesFunction) AsFunc(
+	generationContext *astmodel.CodeGenerationContext, receiver astmodel.TypeName) *dst.FuncDecl {
+
+	krType := astmodel.NewArrayType(astmodel.KubernetesResourceType).AsType(generationContext)
+	krLiteral := astbuilder.NewCompositeLiteralBuilder(krType).Build()
+
+	//iterate through the resourceType versions and add them to the KubernetesResource literal slice
+	for _, resource := range ext.resources {
+		expr := astbuilder.AddrOf(astbuilder.NewCompositeLiteralBuilder(resource.AsType(generationContext)).Build())
+		expr.Decs.Before = dst.NewLine
+		krLiteral.Elts = append(krLiteral.Elts, expr)
+	}
+
+	receiverName := ext.idFactory.CreateReceiver(receiver.Name())
+
+	funcDetails := &astbuilder.FuncDetails{
+		ReceiverIdent: receiverName,
+		ReceiverType:  astbuilder.Dereference(receiver.AsType(generationContext)),
+		Name:          ExtendedResourcesFunctionName,
+		Body:          astbuilder.Statements(astbuilder.Returns(krLiteral)),
+	}
+
+	funcDetails.AddComments("Returns the KubernetesResource slice for Resource versions")
+	funcDetails.AddReturn(krType)
+
+	return funcDetails.DefineFunc()
+}
+
+// Equals returns true if the passed function is equal textus, or false otherwise
+func (ext *GetExtendedResourcesFunction) Equals(f astmodel.Function, _ astmodel.EqualityOverrides) bool {
+	_, ok := f.(*GetExtendedResourcesFunction)
+	// Equality is just based on Type for now
```

Comment:
minor: Not that this is really used, but might not hurt to at least check that the `ext.resources` are the same?

Reply:
Implemented the deep compare on this one.. since resources are sorted beforehand being assigned into the struct, I'm comparing them using `reflect.DeepEqual` in this block.